client/dashboard/sites/settings-site-visibility/privacy-form.tsx

```diff
@@ -44,23 +48,44 @@ const fields: Field< SiteSettings >[] = [
 			},
 		],
 	},
+];
+
+const visibilityForm = {
+	type: 'regular',
+	fields: [ { id: 'wpcom_site_visibility', labelPosition: 'none' } ],
+} satisfies Form;
+
+// This form also has access to `isPrimaryDomainStaging` which isn't a persisted setting, but is data
+// needed to determine whether the checkboxes should be disabled.
+const robotFields: Field< SiteSettings & { isPrimaryDomainStaging: boolean } >[] = [
 	{
 		id: 'wpcom_discourage_search_engines',
-		Edit: 'checkbox',
 		label: __( 'Discourage search engines from indexing this site' ),
 		description: __(
 			'This does not block access to your site — it is up to search engines to honor your request.'
 		),
 		isVisible: ( { wpcom_site_visibility }: SiteSettings ) => wpcom_site_visibility === 'public',
+		Edit: ( { field, onChange, data, hideLabelFromVision } ) => (
+			<CheckboxControl
+				__nextHasNoMarginBottom
+				label={ hideLabelFromVision ? '' : field.label }
+				checked={ data.isPrimaryDomainStaging || field.getValue( { item: data } ) }
```

Comment:
I'm feeling a bit uneasy about this "value hijacking". It means when DataForm finally supports disabled fields, we will still need to use this custom control logic. Perhaps it's OK for this case because it's the case where the form values do not match what get sent to the backend. 😄

Reply:
When DataForms supports disabling, then I think another way to do value hijacking would be to do it when passing the `data` prop. So the `data` prop would look like:

```tsx
	data={ {
		...formData,
		wpcom_discourage_search_engines: isPrimaryDomainStaging || field.getValue( { item: data } )
	} }
```

That way we're not polluting the React state.

Reply:
Does that work? I think the state will still be overridden when we do `{ ...formData, ...edits }` in `handleChange` 🤔